webapp/src/components/SimRecorder.tsx

```diff
@@ -0,0 +1,187 @@
+import * as simulator from "../simulator";
+import * as React from "react"
+import * as screenshot from "../screenshot";
+import { getEditorAsync } from "../app";
+import { SimRecorder, SimRecorderRef, SimRecorderState } from "../../../react-common/components/share/ThumbnailRecorder";
+
+interface SimRecorderRefImpl extends SimRecorderRef {
+    gifAddFrame(data: ImageData, delay?: number): boolean;
+}
+
+export const SimRecorderImpl: SimRecorder = props => {
+    const { onSimRecorderInit } = props;
+
+    const [loanedSimulator, setLoanedSimulator] = React.useState<HTMLElement>(undefined);
+
+    React.useEffect(() => {
+        const sim = simulator.driver.loanSimulator()
+        setLoanedSimulator(sim);
```

Comment:
So React is still a bit of a mystery to me and I may be missing something obvious...but regarding the issue I mentioned in https://github.com/microsoft/pxt/pull/9030#issuecomment-1235770439, I debugged it a little and the issue seems to be that even after calling `setLoanedSimulator` here, `loanedSimulator` still isn't set. My understanding is that this is sort of expected with React since the updates don't happen immediately, but it *still* wasn't set even when the user clicks "Apply" and return is called down in line 46, which seems unexpected. As a result, `unloadSimulator` was never called, and I'm guessing that's why the sim didn't re-appear once the share dialog was closed.

No idea if this is repro'ed on other people's machines, but it's a consistent repro for me.